lib/frizz/remote.rb

```diff
@@ -39,7 +50,9 @@ def delete(remote_file)
     attr_reader :bucket_name, :ignorance, :options
 
     def ignore?(object)
-      ignorance.ignore?(object.key)
+      # for remote files, you don't need to ignore for gzip preference,
+      # since there won't be any gzip files on s3
+      ignorance.ignore_matched_pattern?(object.key)
```

Comment:
The behavior of `Ignorance#ignore?` seems like it would be slightly different for `Local` and  `Remote` class. 

Assuming the prefer_gzip option is true:
For the `Local` class, if we have a `.gz` version of the current file, we would prefer the `.gz` file. However, for the `Remote` class, we would always prefer files without the `.gz` extension, since all files should be uploaded to s3 without the `.gz` extension. 

If all the files were uploaded correctly, `ignore_file_for_gzip_version?` will return false for all files in the Remote class, b/c there shouldn't be any files with the `.gz` extension on s3. This would still be the correct result for the list of remote files. However, the logic behind it doesn't seem accurate for the `Remote` class: we were still searching for  `.gz` files to use in place of non-`.gz` files, when we actually prefer non-`.gz` files in this case.

I'm not sure if that's being too nitpicky, since the result is the same. That's why I was trying to add in some logic to prefer gzip files for the `Local` class, but not prefer gzip files for the `Remote` class.

If we don't need to worry about differentiating between `Local` and `Remote`for gzip file preferences, then I think your suggestion would be great, since all we need is enough info for the local and full path.

If you think we should still differentiate between `Local` and `Remote` classes when it comes to gzip file preferences, maybe we can create 2 instances of `Ignorance` when initializing the `Site` class. For the instance used with `Local`, we pass in the prefer_gzip option value. For the instance use with `Remote`, we pass in false for the prefer_gzip parameter.

```
@local_ignorance = Ignorance.new(@options[:ignore], @options[:prefer_gzip])
@remote_ignorance = Ignorance.new(@options[:ignore], false)

@local = Local.new(path_to_deploy, @local_ignorance, local_options)
@remote = Local.new(path_to_deploy, @remote_ignorance, remote_options)

```